lib/ansible/modules/cloud/amazon/ec2.py

```diff
@@ -673,6 +673,10 @@ def get_reservations(module, ec2, tags=None, state=None, zone=None):
             except:
                 pass
 
+        # if int, we need to convert to a string
+        if isinstance(tags, int):
+            tags = str(tags)
```

Comment:
tag might be 'float' or other types, i recommend:
```
if no isinstance(tags, string_types):
   tags = to_text(tags)
```
string_types and to_text can be imported from module_utils, they are Ansible specific ways to deal with strings across encoding and Python version issues.